snapcraft/models/project.py

```diff
@@ -507,7 +507,7 @@ class Platform(models.Platform):
     build_on: list[SnapArch] | None = pydantic.Field(  # type: ignore[assignment,reportIncompatibleVariableOverride]
         min_items=1, unique_items=True
     )
-    build_for: list[SnapArch] | None = pydantic.Field(  # type: ignore[assignment,reportIncompatibleVariableOverride]
+    build_for: list[SnapArch | Literal["all"]] | None = pydantic.Field(  # type: ignore[assignment,reportIncompatibleVariableOverride]
```

Comment:
out of curiosity, do core22/core20 support `build-for: all`? that is, a string instead of a list? If so, are we also expected to support it?